### PR TITLE
feature(odata-service-inquirer): Adds abap on prem prompts

### PR DESCRIPTION
**Pull request by IainSAP**
#2094 

- Adds Abap on prem prompts

### CODE REVIEW

File: packages/odata-service-inquirer/src/prompts/prompts.ts [hunk 1]
```diff
@@ -109,6 +112,17 @@ async function getDatasourceTypeConditionalQuestions(
         ) as OdataServiceQuestion[])
     );
 
+    // Temp integration into Service Inquirer new system questions
+    conditionalQuestions.push(
```
Comment (IainSAP) [inline packages/odata-service-inquirer/src/prompts/prompts.ts:116]: Add `VSCode` condition, storage of new systems is not supported on BAS

File: packages/odata-service-inquirer/src/prompts/connectionValidator.ts [hunk 2]
```diff
@@ -117,19 +154,107 @@ export class ConnectionValidator {
...
+            this._catalogV2 = (this._serviceProvider as AbapServiceProvider).catalog(ODataVersion.v2);
+        }
+        if (!odataVersion || odataVersion === ODataVersion.v4) {
+            this._catalogV4 = (this._serviceProvider as AbapServiceProvider).catalog(ODataVersion.v4);
+        }
+        try {
+            this._catalogV2 ? await this._catalogV2.listServices() : await this._catalogV4.listServices();
+        } catch (error) {
+            // We will try the v4 catalog if v2 returns a 404
+            if ((error as AxiosError).response?.status === 404 && this._catalogV4) {
```
Comment (cianmSAP) [inline packages/odata-service-inquirer/src/prompts/connectionValidator.ts:225]: What's the reason for only doing this for a 404? 
I have a system I do not have auth to access the V4 catalog and it returns a 403.

Comment (IainSAP) [reply packages/odata-service-inquirer/src/prompts/connectionValidator.ts:225]: 403 is handled elsewhere (auth handling). This is the "ping" to determine the service catalog availability. Some systems do not have v2 catalog enabled (which should return a 404) and so we "fallback" to v4 catalog in this specific case.

Comment (cianmSAP) [reply packages/odata-service-inquirer/src/prompts/connectionValidator.ts:225]: Ok thanks

**Review by cianmSAP — COMMENTED** [review]
Few small typos and queries

File: packages/odata-service-inquirer/package.json [hunk 3]
```diff
@@ -36,10 +36,13 @@
         "@sap-ux/inquirer-common": "workspace:*",
         "@sap-ux/logger": "workspace:*",
         "@sap-ux/project-access": "workspace:*",
+        "@sap-ux/project-input-validator": "workspace:*",
+        "@sap-ux/store": "workspace:*",
```
[the remark this answers is not in the document]
Comment (IainSAP) [reply packages/odata-service-inquirer/package.json:40]: Thanks, yep, should be a dev dep now.

[the remark this answers is not in the document]
Comment (IainSAP) [reply packages/odata-service-inquirer/package.json:40]: The classes `BackendSystem` and `SystemService` are both required so there is no alternative.

File: packages/odata-service-inquirer/src/translations/odata-service-inquirer.i18n.json [hunk 4]
```diff
@@ -60,7 +60,47 @@
             "warningCertificateValidationDisabled": "User has disabled certificate validation",
             "annotationsNotFound": "Annotations not found for specified service"
         },
-        "nonUIServiceTypeWarningMessage": "Please note that {{serviceTypeDesc}} services are not intended to be used for the generation of SAP Fiori UI applications"
+        "warnings": {
+            "nonUIServiceTypeWarningMessage": "Please note that {{serviceType}} services are not intended to be used for the generation of SAP Fiori UI applications",
+            "noServicesAvailable": "No services available for the selected system, see logs for furthur details.",
```
[the remark this answers is not in the document]
Comment (IainSAP) [reply packages/odata-service-inquirer/src/translations/odata-service-inquirer.i18n.json]: Thx done.

**Review by cianmSAP — APPROVED** [review]
New logic looks good for abap on prem prompts
Comments and questions addressed
Have not tested
Changeset

File: packages/odata-service-inquirer/src/prompts/connectionValidator.ts (hunk 2 above)
Comment (devinea) [reply packages/odata-service-inquirer/src/prompts/connectionValidator.ts:225]: Thanks for including this.

**Review by devinea — APPROVED** [review]
Code looks clean and covered by tests. 
changeset ✅ 
haven't tested locally.

**Pull request merged**